### ISSUE TITLE
BadJpqlGrammarException with attribute labelled 'type'

### ISSUE THREAD

**Issue by gpolet** [spring-projects/spring-data-jpa#3062]
Similarly to https://github.com/spring-projects/spring-data-jpa/issues/3028, sping-data-jpa throws BadJpqlGrammarException: Line 1:33 no viable alternative at input 'e.type'; Bad JPQL grammar [select e from Employee e where e.type = :_type]

```
public interface EmployeeRepository extends JpaRepository<Employee, Long> {

    @Query("select e from Employee e where e.type = :_type")
    List<Employee> failingQuery(@Param("_type") String name);
}
```
Prior sping-data-jpa 3.1.0 query was working (it also seems that named parameters that are matching reserved keywords, eg: type, date, from, ..., also make the parsing crash, unlike allowed prior v3.1.0)

**Comment by gregturn** [comment]
I've confirmed this for JPQL parser (HQL parsing handles this just fine).

**Comment by gregturn** [comment]
Merged to `main` and backported to `3.1.x`
